diff --git a/dist/src/content-for.js b/dist/src/content-for.js
index b7541151131e62c64e83bbd96ac486679d43d0db..f22df7c9a051502a1413c25fe15ffa0c8aaee996 100644
--- a/dist/src/content-for.js
+++ b/dist/src/content-for.js
@@ -9,11 +9,20 @@ export function contentFor() {
         enforce: 'pre',
         transformIndexHtml(html, { path }) {
             let config = readJSONSync(join(locateEmbroiderWorkingDir(process.cwd()), 'content-for.json'));
+            
             let contentsForConfig = config[path];
+
+            
+
             for (const [contentType, htmlContent] of Object.entries(contentsForConfig)) {
                 html = html.replace(`{{content-for "${contentType}"}}`, `${htmlContent}`);
                 html = html.replace(`{{content-for '${contentType}'}}`, `${htmlContent}`);
             }
+
+            if (config['/index.html'].head.includes('production')) {
+                html = html.replaceAll('assets/', 'ember-common-ui/assets/');
+            }
+            
             return html;
         },
     };
